File: EconRentCarWeb/EconRentCar.DataModel/Vehiculo.cs
using EconRentCar.Common;
using EconRentCar.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EconRentCar.DataModel
{
    public class Vehiculo: IEntityBase
    {
        public Vehiculo()
        {
            Rentas = new HashSet<Renta>();
        }
        public Guid Id { get; set; }
        [Required]
        public string Placa { get; set; }
        [Required]
        public string Descripcion { get; set; }
        [Required]
        public string NoChasis { get; set; }
        [Required]
        public string NoMotor { get; set; }
        [Required]
        public EstadoVehiculo EstadoVehiculo { get; set; }
        public int ModeloId { get; set; }
        public virtual Modelo Modelo { get; set; }
        public int TipoVehiculoId { get; set; }
        public virtual TipoVehiculo TipoVehiculo { get; set; }
        public int TipoCombustibleId { get; set; }
        public virtual TipoCombustible TipoCombustible { get; set; }

        public virtual ICollection<Renta> Rentas { get; set; }

    }
}

File: EconRentCarWeb/EconRentCarClient/src/app/shared/models/vehiculo.ts
import { BaseData } from '../dynamic-crud/models';
import { EstadoVehiculo } from './enums';
import { Modelo } from './modelo';
import { TipoVehiculo } from './tipo-vehiculo';
import { TipoCombustible } from './tipo-combustible';
import { Renta } from './renta';

export class Vehiculo implements BaseData {
    public Id = '00000000-0000-0000-0000-000000000000';
    public Placa = '';
    public Descripcion = '';
    public NoChasis = '';
    public NoMotor = '';
    public EstadoVehiculo = EstadoVehiculo.Nuevo;
    public ModeloId = '00000000-0000-0000-0000-000000000000';
    public Modelo: Modelo;
    public TipoVehiculoId = '00000000-0000-0000-0000-000000000000';
    public TipoVehiculo: TipoVehiculo;
    public TipoCombustibleId = '00000000-0000-0000-0000-000000000000';
    public TipoCombustible: TipoCombustible;
    public Rentas: Renta[];
}
